ui/page/components/items_scroll.go

```diff
@@ -67,47 +70,48 @@ func (s *Scroll[T]) FetchScrollData(isReverse bool, window app.WindowNavigator)
 		s.offset -= s.pageSize
 	}
 	s.mu.Unlock()
-
-	s.fetchScrollData(isReverse, window)
+	// set isReverse to default false as callers of this method are not
+	// perform a reverse scroll action
+	s.fetchScrollData(false, isReset, window)
 }
 
 // fetchScrollData fetchs the scroll data and manages data returned depending on
 // on the up or downwards scrollbar movement. Unless the data fetched is less than
 // the page, all the old data is replaced by the new fetched data making it
 // easier and smoother to scroll on the UI. At the end of the function call
 // a window reload is triggered.
-func (s *Scroll[T]) fetchScrollData(isReverse bool, window app.WindowNavigator) {
+func (s *Scroll[T]) fetchScrollData(isReverse, isReset bool, window app.WindowNavigator) {
 	s.mu.Lock()
 
+	if isReset {
+		// resets the values for use on the next iteration.
+		s.resetList()
+	}
+
+	// reset loaded items if we have loaded all items and we are scrolling back to the top
+	if isReverse && s.loadedAllItems {
+		s.loadedAllItems = false
+	}
```

Comment:
Why? If all items were already loaded, scrolling back up doesn't delete the items that were already loaded, does it?